c_tests/tests/rel_path.c

```diff
@@ -0,0 +1,58 @@
+// Compiler:
+// Run-time:
+//   env-var: YK_PRINT_IR=1
+//   stderr:
+//   stderr:
+//     define internal void @__yk_compiled_trace_0(i32* %0) {
+//        store i32 2, i32* %0, align 4, !tbaa !0
+//        ret void
+//     }
+//     ...
+
+// Check that running a traced binary via a relative path works.
+//
+// FIXME An optimising compiler can remove all of the code between start/stop
+// tracing.
+
+#include <assert.h>
+#include <err.h>
+#include <libgen.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  if (*argv[0] == '/') {
+    // Reinvoke ourself with a relative path.
```

Comment:
This is wonderfully evil :p I don't know if I'd have thought of this!

Reply:
It took a while to invent :P